synapse/storage/schema/main/delta/58/05cache_instance.sql.postgres

```diff
@@ -28,3 +28,8 @@ CREATE TABLE cache_invalidation_stream_by_instance (
 CREATE UNIQUE INDEX cache_invalidation_stream_by_instance_id ON cache_invalidation_stream_by_instance(stream_id);
 
 CREATE SEQUENCE cache_invalidation_stream_seq;
+
+-- The sequence needs to begin at 2 because a bunch of code assumes that
+-- get_next_id_txn will return values >= 2, cf this comment:
+-- https://github.com/matrix-org/synapse/blob/b93bd95e8ab64d27ae26841020f62ee61272a5f2/synapse/storage/util/id_generators.py#L344
+SELECT setval('cache_invalidation_stream_seq', 1);
```

Comment:
Does this overwrite the value in an existing databases? Does that matter?

Do we need to do some `COALESCE(MAX(...))` thing like we do for other sequences?

https://github.com/matrix-org/synapse/blob/fa2f3d8d0c915c6ebd84ef53526ac04c91fd4a19/synapse/storage/schema/main/delta/58/14events_instance_name.sql.postgres#L18-L20

Reply:
It shouldn't override anything, since the migration step is only run once, and the `setval` is done right after the table creation. However that makes me think there is one case where it would stay to 0: if the server has been created before this change but replication was 0, when turning on it would still be at 0.

I am hence going to change my code to do this `COALESCE(MAX(...))` in a new migration step that, unlike this one, will also run on upgrade, not only creation of a new server.